include/psa/crypto.h

```diff
@@ -3759,6 +3759,418 @@ psa_status_t psa_generate_key(const psa_key_attributes_t *attributes,
 
 /**@}*/
 
+
+/** \defgroup Password-authenticated key agreement
+ * @{
+ */
+
+/** The type of the data strucure for PAKE cipher suites.
+ *
+ * This is an implementation-defined \c struct. Applications should not
+ * make any assumptions about the content of this structure except
+ * as directed by the documentation of a specific implementation.
+ */
+typedef struct psa_pake_cipher_suite_s psa_pake_cipher_suite_t;
+
+/** Construct a cipher suite for a password-authenticated key exchange.
+ *
+ * \param primitive          The primitive used in the cipher suite.
+ * \param hash               The hash involved in the cipher suite.
+ *                           (`PSA_ALG_XXX` values of type ::psa_algorithm_t
+ *                           such that #PSA_ALG_IS_HASH(\c alg) is true.)
+ * \param algorithm1         Additional algorithm if needed in the cipher suite,
+ *                           0 otherwise.
+ * \param bits1              A bit size qualifier if needed for \p algorithm1,
+ *                           0 otherwise.
+ * \param algorithm2         Additional algorithm if needed in the cipher suite,
+ *                           0 otherwise.
+ * \param bits2              A bit size qualifier if needed for \p algorithm2,
+ *                           0 otherwise.
+ * \param options            Additional options to be included with the cipher
+ *                           suite if needed, 0 otherwise.
+ *
+ * Refer to the documentation of individual PAKE algorithm types (`PSA_ALG_XXX`
+ * values of type ::psa_algorithm_t such that #PSA_ALG_IS_PAKE(\c alg) is true)
+ * for more information.
+ *
+ * \retval The constructed cipher suite.
+ */
+static psa_pake_cipher_suite_t psa_pake_cipher_suite(
```

Comment:
EC J-PAKE is quite flexible regarding cipher suites, but many PAKEs only have a handful of them. For these it would be convenient to have the ability to define constants for cipher suites that can be passed to the setup function directly.

For example:

```c
#define SPAKE2_P256_SHA256_HMAC                       \
psa_pake_cipher_suite(                                \
    PSA_PAKE_SPAKE2,                                  \
    PSA_PAKE_PRIMITIVE(                               \
        PSA_PAKE_PRIMITIVE_TYPE_CURVE,                \
        PSA_ECC_FAMILY_SECP_R1,                       \
        256),                                         \
    PSA_ALG_SHA_256,                                  \
    PSA_ALG_HMAC_BASE,                                \
    0, 0, 0, 0)
```
(In this example I already incorporated your suggestion below - https://github.com/ARMmbed/mbedtls/pull/4317#discussion_r613476966)

This could be useful for J-PAKE as well since protocols using it have to agree on it a-priori or negotiate it. Either way there is a limited set of cipher suites and it would make sense to define them as such constants.

Reply:
I have reworked this part as we agreed to use getters and setters. (We can add convenience macros later.) I also only added setter/getter functions for the attributes needed by JPAKE. Further functions can be added when needed as the structure is implementation defined.